pandas/plotting/_matplotlib/core.py

```diff
@@ -505,16 +505,35 @@ def _adorn_subplots(self):
             )
 
         for ax in self.axes:
+
             if self.yticks is not None:
                 ax.set_yticks(self.yticks)
 
             if self.xticks is not None:
                 ax.set_xticks(self.xticks)
 
             if self.ylim is not None:
+                # Addressing issue 40781
+                for elem in self.ylim:
+                    if not is_float(elem) and not is_integer(elem):
+                        raise ValueError(
+                            "`ylim` expects either float"
+                            " or integer datatype.\n"
+                            "`ylim` has unsupported dtype"
+                            f"of {type(elem)} with value {elem}.\n"
+                        )
                 ax.set_ylim(self.ylim)
 
             if self.xlim is not None:
+                # Addressing issue #40781
+                for elem in self.xlim:
+                    if not is_float(elem) and not is_integer(elem):
+                        raise ValueError(
+                            "`xlim` expects either float"
+                            " or integer datatype.\n"
+                            "`xlim` has unsupported dtype"
+                            f" {type(elem)} with value {elem}.\n"
+                        )
```

Comment:
This logic looks the same for both `xlim` and `ylim`, can you find a way to share it?

Reply:
Addressing this with a lambda function

Reply:
Sharing the message is nice, but would be great to also share the validation logic itself (eg the iteration and type checking). Also, I think generally it's discouraged to assign an anonymous function to a variable, if doing that better to be explicit and write a function (which could also be inline).